cli/main.go

```diff
@@ -0,0 +1,176 @@
+}
+
+func getBeaconClient(beacon_uri string) (BeaconClient, error) {
+	beaconClient, _, err := NewBeaconClient(beacon_uri)
+	return beaconClient, err
+}
+
+func lastCheckpointedForEigenpod(eigenpodAddress string, client *ethclient.Client) uint64 {
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), client)
+	PanicOnError(err)
+
+	timestamp, err := eigenPod.LastCheckpointTimestamp(nil)
+	PanicOnError(err)
+	return timestamp
+}
+
+func computeSlotImmediatelyPriorToTimestamp(timestampSeconds uint64, genesis time.Time) uint64 {
+	return uint64(math.Floor(float64(timestampSeconds)-float64(genesis.Unix())) / 12)
+}
+
+// search through beacon state for validators whose withdrawal address is set to eigenpod.
+func findAllValidatorsForEigenpod(eigenpodPubKey string, beaconState *spec.VersionedBeaconState) []ValidatorWithIndex {
+	allValidators, err := beaconState.Validators()
+	PanicOnError(err)
+
+	expectedCredentials := sha256.Sum256([]byte(eigenpodPubKey))
+	expectedCredentials[0] = 0 // the first byte of withdrawal credentials is set to 0.
+
+	var outputValidators []ValidatorWithIndex = []ValidatorWithIndex{}
+	var i uint64 = 0
+	maxValidators := uint64(len(allValidators))
+	for i = 0; i < maxValidators; i++ {
+		validator := allValidators[i]
+		if validator == nil {
+			continue
+		}
+		if bytes.Equal(expectedCredentials[:], validator.WithdrawalCredentials) {
+			outputValidators = append(outputValidators, ValidatorWithIndex{
+				Validator: validator,
+				Index:     i,
+			})
+		}
+	}
+	return outputValidators
+}
+
+func getOnchainValidatorInfo(client *ethclient.Client, eigenpodAddress string, allValidators []ValidatorWithIndex) []onchain.IEigenPodValidatorInfo {
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), client)
+	PanicOnError(err)
+
+	var validatorInfo []onchain.IEigenPodValidatorInfo = []onchain.IEigenPodValidatorInfo{}
+
+	// TODO: batch/multicall
+	for i := 0; i < len(allValidators); i++ {
+		pubKeyHash := sha256.Sum256((allValidators[i]).Validator.PublicKey[:])
+		info, err := eigenPod.ValidatorPubkeyHashToInfo(nil, pubKeyHash)
+		PanicOnError(err)
+		validatorInfo = append(validatorInfo, info)
+	}
+
+	return validatorInfo
+}
+
+// Stub for the execute function
+func execute(ctx context.Context, eigenpodAddress, beacon_node_uri, node string, out *string) {
+	eth, err := ethclient.Dial(node)
+	if err != nil {
+		fmt.Printf("ERROR: Invalid node - Failed to connect to `%s`.\n\n", node)
+		PanicOnError(err)
+	}
+	beaconClient, err := getBeaconClient(beacon_node_uri)
+	PanicOnError(err)
+
+	lastCheckpoint := lastCheckpointedForEigenpod(eigenpodAddress, eth)
+
+	// fetch the beacon state which corresponds to the slot immediately prior to this timestamp.
+	genesis, err := beaconClient.GetChainGenesisTime(ctx)
+	PanicOnError(err)
+
+	slot := computeSlotImmediatelyPriorToTimestamp(lastCheckpoint, genesis)
+	header, err := beaconClient.GetBeaconHeader(ctx, strconv.FormatUint(slot, 10))
```

Comment:
so for this, instead of calculating the slot (this is hard because you need the latest non missed slot), just

1. get the current checkpoint https://github.com/Layr-Labs/eigenlayer-contracts/blob/d148952a2942a97a218a2ab70f9b9f1792796081/src/contracts/interfaces/IEigenPod.sol#L224
2. parse the current checkpoint's block root https://github.com/Layr-Labs/eigenlayer-contracts/blob/d148952a2942a97a218a2ab70f9b9f1792796081/src/contracts/interfaces/IEigenPod.sol#L39
3. use the block root as the second param in `GetBeaconHeader`